test/e2e/scale.go

```diff
@@ -0,0 +1,213 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"fmt"
+	"net/http"
+	"testing"
+	"time"
+
+	pkgTest "github.com/knative/pkg/test"
+	"github.com/knative/pkg/test/logging"
+	"github.com/knative/serving/pkg/apis/serving/v1alpha1"
+	serviceresourcenames "github.com/knative/serving/pkg/reconciler/v1alpha1/service/resources/names"
+	"github.com/knative/serving/test"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+
+	"github.com/knative/serving/pkg/pool"
+	. "github.com/knative/serving/pkg/reconciler/v1alpha1/testing"
+)
+
+// Latencies is an interface for providing mechanisms for recording timings
+// for the parts of the scale test.
+type Latencies interface {
+	// Add takes the name of this measurement and the time at which it began.
+	// This should be called at the moment of completion, so that duration may
+	// be computed with `time.Since(start)`.  We use this signature to that this
+	// function is suitable for use in a `defer`.
+	Add(name string, start time.Time)
+}
+
+func ScaleToWithin(t *testing.T, logger *logging.BaseLogger, scale int, duration time.Duration, latencies Latencies) {
+	clients := Setup(t)
+
+	cleanupCh := make(chan test.ResourceNames, scale)
+	defer close(cleanupCh)
+
+	fopt := []ServiceOption{
+		// We set a small resource alloc so that we can pack more pods into the cluster.
+		func(svc *v1alpha1.Service) {
+			svc.Spec.RunLatest.Configuration.RevisionTemplate.Spec.Container.Resources = corev1.ResourceRequirements{
+				Limits: corev1.ResourceList{
```

Comment:
I am not sure if the PR  where those are defined as global test constants is already merged, if it is -- we should probably use that.

Reply:
it hasn't landed yet.